checks.d/fluentd.py

```diff
@@ -40,11 +41,13 @@ def check(self, instance):
             status = r.json()
 
             for p in status['plugins']:
+                tag = "%s:%s" % (tag_by, p.get(tag_by))
                 for m in self.GAUGES:
                     if p.get(m) is None:
                         continue
-                    if p.get('plugin_id') in plugin_ids:
-                        self.gauge('fluentd.%s' % (m), p.get(m), ["plugin_id:%s" % p.get('plugin_id')])
+                    # Filter unspecified plugins to keep backward compatibility.
+                    if tag_by == 'type' or len(plugin_ids) == 0 or p.get('plugin_id') in plugin_ids:
```

Comment:
I am not sure to understand why we are checking the value of `tag_by` here. I would just write:
```python
 if len(plugin_ids) == 0 or p.get('plugin_id') in plugin_ids:
```
Am I missing something ?